src/VecSim/algorithms/brute_force/brute_force.cpp

```diff
@@ -154,6 +154,11 @@ int BruteForceIndex::deleteVector(size_t label) {
 
 size_t BruteForceIndex::indexSize() const { return this->count; }
 
+int BruteForceIndex::resolveParams(VecSimRawParam *rparams, VecSimQueryParams *qparams) {
+    bzero(qparams, sizeof(VecSimQueryParams));
```

Comment:
validate NULL
also, what about the index type if you are setting to zeros?

Reply:
VecSimQueryParams doesn't hold type field. Each index is looking for its part in the union.
Right now VecSimQueryParams is just size_t for hnsw.